helm/resource_release.go

```diff
@@ -1508,3 +1524,21 @@ func valuesKnown(d *schema.ResourceDiff) bool {
 	}
 	return true
 }
+
+func useChartVersion(chart string, repo string) bool {
+	// checks if chart is a URL or OCI registry
+
+	if url, err := http.Get(chart); err == nil && !registry.IsOCI(chart) {
+		return url.Header.Get("Content-Type") == "binary/octet-stream"
```

Comment:
Thank you for the comments! I seemed to have forgot to include the checks within `useChartVersion`.  I added more specific checks before the chart gets downloaded within `getChart()`

```hcl
func getChart(d resourceGetter, m *Meta, name string, cpo *action.ChartPathOptions) (*chart.Chart, string, error) {
	//Load function blows up if accessed concurrently
	m.Lock()
	defer m.Unlock()

	// Checks if chart is a URL; checks if it's a valid URL to a .tgz file of the chart
	url, err := http.Get(name)
	if err == nil {
		contentType := url.Header.Get("Content-Type")

		if contentType != "binary/octet-stream" && contentType != "application/x-gzip" && contentType != "application/x-compressed-tar" {
			return nil, "", fmt.Errorf("Not an absolute URL to the .tgz of the Chart")
		}
	}
```

1. Based on the acceptance tests, the only Content-Types that can show up when using an archive would be : `octet-stream`, `x-gzip`, `x-compressed-tar`
2. By the time we reach `useChartVersion` The headers attribute would already be populated since this would be done after downloading the chart in `getChart()`

I'm wondering if we need to still include the check within `useChartVersion` since we are doing the check of whether or not the link is a valid URL Archive link within `getChart`, it will not allow the use of a link in the `chart` attribute if its not a valid archive file. Thoughts on this? I can still include the extra check in if it makes sense to add.